pennylane/templates/subroutines/reflection.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This submodule contains the template for the Reflection operation.
+"""
+
+import numpy as np
+import pennylane as qml
+from pennylane.operation import Operation
+from pennylane.ops import SymbolicOp
+
+
+class Reflection(SymbolicOp, Operation):
```

Comment:
Because a user could write:

```python
qml.Reflection(qml.Hadamard(0), ....)
```

SymbolicOp makes sure that qml.Hadamard is not queued accidentally